### ISSUE TITLE
size mismatch 

### ISSUE THREAD

**Issue by defields923** [kamilkp/angular-vs-repeat#202]
I am having an issue since I upgraded from version 1.x to the latest. I have a list of elements all the same size within a ul that has a height of calc(100% - 31px). vs-repeat appears to be working, but I get that size mismatch error in the console, and many more are logged when I scroll. I tried adjusting the size option like so 'vs-repeat="{size: ...}"' to try and match the expected size, but I can only get as close as .000000001... to it no matter how much I try.

Was this an issue in the earlier version that just wasn't logged? What should I do to correct this, or is there at least a way of turning the logging off?

**Comment by XueDong369** [comment]
I have the same problem with you!
`vsRepeat: size mismatch. Expected size 0px whereas actual size is 287px. Fix vsSize on element: <ul class=​"common-list-content" vs-repeat=​"{size:​ 36}​">​…​</ul>​
`

**Comment by tje3d** [comment]
You cant get rid of size mismatch, because if you set fixed size for your vs-repeat, you always have this warning, and if you don't set it, You don't have any scroll ( because the size of vs-repeat should not be same as sum of all elements height )

**Comment by matortheeternal** [comment]
I've experienced this issue on v1.1.11 as well as v2.0.9.  My temporary solution is to downgrade back to v1.1.7.  I tried setting autoresize to true in the vs-repeat attribute options, but this did not fix the issue.  Hardcoding a size does not make sense for my application where the window the resizable.  The size should just be computed based on the element's size by default like it was in v1.1.7.

**Comment by XueDong369** [comment]
I am using v2.0.9 now, but this problem always not fixed yet!

**Comment by tje3d** [comment]
My problems is when i apply a filter to my ng-repeat, and the height of my items goes lower than vs-repeat height, i always have `size mismatch` warning.
a hackfix is to calculate vs-repeat size when a filter applies but this is not the correct way.

**Comment by XueDong369** [comment]
@tje3d But it will cause the performance problem if the warning log always repeat ...

**Comment by tje3d** [comment]
@XueDong369 please check my pull request #206
it will prevent warning when there is no overflow scroll

**Comment by XueDong369** [comment]
@tje3d Yes, I've changed the local code to #206 temporarily, then it won't console the warning info. But, my project will run in Docker and it will run 'bower install' in Docker every time when I deploy it. I hope #206 will be merged ASAP. ~_~

**Comment by XueDong369** [comment]
@tje3d Yes, I've changed the local code to #206 temporarily, then it won't console the warning info. Thanks!  But, my project will run in Docker and it will run 'bower install' in Docker every time when I deploy it. I hope #206 will be merged ASAP.  ~_~

**Comment by tcrite** [comment]
I had this same issue. Fixed it by inspecting every element in row and taking into account margins. Also had an inner element whose margin expanded pass the parent container causing the calculated size to be off.

**Comment by tcrite** [comment]
also, hiding the warning is just a mask...the underlying issue of the size being off will still end up in performance issues.

**Comment by DanielRuf** [comment]
Any news?

**Comment by ibrychgo** [comment]
Same here. Getting this on list of md-cards (angularjs material). Was hoping for an easier solution to md-virtual-repeat.

**Comment by erceth** [comment]
I was able to work around the warning spam by removing the margin from my list elements. I think it was due to the fact that child margin doesn't affect parent height (https://stackoverflow.com/q/27829250/3038518). Perhaps a future fix for this bug would involve that information somehow.

**Comment by lslee714** [comment]
Love 3 year old issues

**Comment by kerryj89** [comment]
I was happy to see the undocumented option `vs-repeat="{ sizeMismatchWarningsSilenced: true }"`.  The mismatch is about 100 pixels off and does no harm that I can see, so I turned the warning off.

**Comment by kerryj89** [comment]
My previous comment (since deleted) was a false positive, but I am able to hide the warnings with the below snippet.  I'm not well versed in AngularJS but piecing together some knowledge I was able to override (set?) `sizeMismatchWarningsSilenced` to `true` via decorator.  I tried the more common way to config with `angular.module('whatever', ['vs-repeat']).config(['vsRepeatConfig', function(vsRepeatConfig) {...})` but couldn't get it to work.  Maybe someone more knowledgeable in AngularJS can see the error in my way.

```
angular.module('whatever', ['vs-repeat'])
    .config(['$provide', function($provide) {
        $provide.decorator('vsRepeatConfig', function($delegate) {
            return {
                ...$delegate,
                sizeMismatchWarningsSilenced: true
            };
        });
    }])
```
